sql/src/test/java/io/crate/expression/scalar/string/StringPaddingFunctionTest.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import io.crate.expression.scalar.AbstractScalarFunctionsTest;
+import io.crate.expression.symbol.Literal;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class StringPaddingFunctionTest extends AbstractScalarFunctionsTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void test_evaluate_lpad_function() throws Exception {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("len argument exceeds predefined limit of 1000");
```

Comment:
If `expectedException` is used, the first statement that fails will end the test-case, so everything after the next evaluate is not executed.

I think it would also be nice to split the different test cases into multiple methods, where each method verifies a concrete test scenario. 
Otherwise it is a) harder to debug if e.g. the 7th assertion fails and b) harder to understand later on what exactly was the intention of the test if we have to make adaptions due to refactorings.

A good thing to ask yourself is: Would it be possible to re-create the test body given the test method name.

Reply:
agreed, this has been done.